### PR TITLE
Add Blazor enhanced nav events

### LINKED ISSUE #34086: Blazor `enhancednavigationstart` and `enhancednavigationend` events
### Description

Per https://github.com/dotnet/aspnetcore/issues/58630.

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/static-server-rendering?view=aspnetcore-8.0

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/javascript-interoperability/static-server-rendering.md

### Document ID

c991c83d-dc0b-1962-2ef2-d7da0a7e56f8

### Article author

@guardrex

[Related Issues](https://github.com/dotnet/AspNetCore.Docs/issues?q=is%3Aissue+is%3Aopen+c991c83d-dc0b-1962-2ef2-d7da0a7e56f8)

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #34086
Addresses #31909

Per https://github.com/dotnet/aspnetcore/issues/58630.

They seem straightforward enough. I didn't see tests for the new events in the framework. Do you have any specific example(s) that you'd like to show for them?

Also, I'm not sure if using the word "load" in the way that I have for the description is correct for `enhancedload`. 👀

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/blazor/javascript-interoperability/static-server-rendering.md](https://github.com/dotnet/AspNetCore.Docs/blob/777d6a978d79a8174d2df8324cd37282dcd58b5a/aspnetcore/blazor/javascript-interoperability/static-server-rendering.md) | [aspnetcore/blazor/javascript-interoperability/static-server-rendering](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/static-server-rendering?branch=pr-en-us-34101) |

### CODE REVIEW

**Review by MackinnonBuck — APPROVED** [review]

File: aspnetcore/blazor/javascript-interoperability/static-server-rendering.md [hunk 1]
```diff
@@ -20,7 +20,47 @@ This article explains how to load JavaScript (JS) in a Blazor Web App with stati
...
+  blazor.addEventListener("enhancednavigationstart", {CALLBACK});
+  ```
+
+* Enhanced navigation end (`enhancednavigationend`) triggers a callback after an enhanced navigation occurs:
+
+  ```javascript
+  blazor.addEventListener("enhancednavigationend", {CALLBACK});
+  ```
+
+* Enhanced navigation page load (`enhancedload`) triggers a callback when an enhanced page loads:
```
Comment (MackinnonBuck) [inline aspnetcore/blazor/javascript-interoperability/static-server-rendering.md:43]: This gets called each time the page updates due to an enhanced navigation. This includes streaming updates.

**Comment by MackinnonBuck** [comment]
Sorry, I should have commented first, then approved. My bad!

**Comment by MackinnonBuck** [comment]
> They seem straightforward enough. I didn't see tests for the new events in the framework. Do you have any specific example(s) that you'd like to show for them?

Not in particular - I think it's enough to call out that they exist. I actually added them because we had a need for them in the framework itself.

**Comment by guardrex** [comment]
I'll submit a patch, but I'm on the treadmill running off a lot of 🦃🦃🦃🦃😆. I'll update this by EOD before it goes live.

**Pull request merged**